### PR TITLE
OpenMapTiles

### PR DESCRIPTION
**Pull request by boldtrn**
This PR looks at using OpenMapTiles via tilehosting.com.

For transforming the styles I used their [Schema](https://openmaptiles.org/schema) and their [debugging page](https://openmaptiles.org/inspect).

So far, the basics are working. Still quite a lot of things are missing.

For some reason I cannot use any style that uses min-zooms >= 15. Everything I tried with min-zoom=15 or above simply did not render. Maybe because max-zoom of the tiles is 16?

Overall the geojson tiles from tilehosting.com feel pretty slow. Also I am not really convinced by their schema as it makes things difficult and as you can see requires a completely changed render theme.

Some screenshots:
openmaptiles-vtm-stuttgart
openmaptiles-vtm
bw

### CODE REVIEW

**Comment by devemux86** [comment]
Thanks, very nice work!

BTW there are already [OpenMapTilesGeojsonTileSource](https://github.com/mapsforge/vtm/blob/master/vtm-json/src/org/oscim/tiling/source/geojson/OpenMapTilesGeojsonTileSource.java) and [OpenMapTilesGeojsonTest](https://github.com/mapsforge/vtm/blob/master/vtm-playground/src/org/oscim/test/OpenMapTilesGeojsonTest.java) classes, also [OpenMapTilesGeojsonMapActivity](https://github.com/mapsforge/vtm/blob/master/vtm-android-example/src/org/oscim/android/test/OpenMapTilesGeojsonMapActivity.java). Can you move any improvements in them?

**Comment by devemux86** [comment]
Indeed their GeoJSON tiles are slow compared with Mapzen.

**Comment by devemux86** [comment]
Thanks!
Squashed and merged via 5fc42268a463333cc7a7b13e458e9c07a09fbd6c, removing the duplicated classes.

File: vtm-json/src/org/oscim/tiling/source/geojson/OpenmaptilesGeojsonTileSource.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+public class OpenmaptilesGeojsonTileSource extends GeojsonTileSource {
+
+    private final static String DEFAULT_URL = "https://free.tilehosting.com/data/v3";
+    private final static String DEFAULT_PATH = "/{Z}/{X}/{Y}.geojson";
+
+    public static class Builder<T extends Builder<T>> extends UrlTileSource.Builder<T> {
+        private String locale = "";
+
+        public Builder() {
+            super(DEFAULT_URL, DEFAULT_PATH, 1, 16);
```
Comment (devemux86) [inline vtm-json/src/org/oscim/tiling/source/geojson/OpenmaptilesGeojsonTileSource.java]: You are right about the max zoom, 17 doesn't seem having tiles.

**Comment by boldtrn** [comment]
Sorry for duplicating the classes, I must have missed that :(.

I would like to add, that the current state is only basically usable. There are still many things in the render theme missing. And maybe even some code improvements might be necessary regarding overzooming to properly render house numbers and some pois.

I probably won't have the time to improve this in the near future.

**Comment by devemux86** [comment]
Thanks again for all the work!

> And maybe even some code improvements might be necessary regarding overzooming to properly render house numbers and some pois.

That would probably need #393 to work properly.

**Pull request closed without merging**